### ISSUE TITLE
Issues with Giac/Sage/ILV

### ISSUE THREAD

**Issue by DRL** [DRL/gIMble#11]
Giac/Sage *sometimes* crashes when doing ILPs 

- [x] Specify Time/Dummy variable as REAL 
- [x] Replicate error
- [x] wrap ILPs in try/except so it doesn't die unexpectedly
- [x] write logging function to output equations (?)
- [ ] TBD

**Comment by DRL** [comment]
```
./gIMble optimise -z ~/data/gimble.heliconius.intergenic.testing_chr02_chr09_chr18.z -c ~/data/debug_giac/gimble.model.s_A_B.n_1_1.M_B_A.J_A_B.ini -b -m ~/data/debug_giac/gimble.model.s_A_B.n_1_1.M_B_A.J_A_B.tsv -t 4,1
[+] Config file validated.
[+] Generated all parameter combinations.
[+] Reading model '/Users/dlaetsch/data/debug_giac/gimble.model.s_A_B.n_1_1.M_B_A.J_A_B.tsv'
[=] ==================================================
[+] Initiating model ...
[%] Building equations: 100%|█████████████████████████████████████| 95/95 [00:00<00:00, 2135.41it/s]
[%] Placing mutations: 100%|████████████████████████████████████████| 45/45 [00:09<00:00,  4.93it/s]
[+] Bundled equations along 45 mutation tuples
[%] Accounting for marginal probabilities: 100%|█████████████████| 112/112 [00:00<00:00, 234.18it/s]
[+] Generated equations for 112 mutation tuples
[+] Optimization starting for specified point.
iteration 	Ne_A	Ne_A_B	me_B_A	T	 lnCL
Traceback (most recent call last):
  File "./gIMble", line 8, in <module>
    main(os.path.dirname(os.path.realpath(__file__)))
  File "/Users/dlaetsch/git/gIMble/cli/interface.py", line 93, in main
    optimise.main(params)
  File "/Users/dlaetsch/git/gIMble/cli/optimise.py", line 81, in main
    parameterObj
  File "/Users/dlaetsch/git/gIMble/lib/math.py", line 604, in optimize_parameters
    allResults.append(run_single_optimiz(startPos, lower, upper, specified_objective_function, parameterObj.max_eval, parameterObj.xtol_rel, parameterObj.ftol_rel))
  File "/Users/dlaetsch/git/gIMble/lib/math.py", line 229, in run_single_optimiz
    optimum = opt.optimize(p0)
  File "/Users/dlaetsch/miniconda3/envs/gimble/lib/python3.7/site-packages/nlopt.py", line 334, in optimize
    return _nlopt.opt_optimize(self, *args)
  File "/Users/dlaetsch/git/gIMble/lib/math.py", line 203, in objective_function
    ETPs = equationSystemObj.calculate_ETPs((rates, split_time, threads, verbose))
  File "/Users/dlaetsch/git/gIMble/lib/math.py", line 534, in calculate_ETPs
    assert math.isclose(np.sum(ETPs.flatten()), 1, rel_tol=1e-5), "[-] sum(ETPs) != 1 (rel_tol=1e-5)"
```

**Comment by GertjanBisschop** [comment]
inverse laplace tranform algorithm is now `maxima` (which is default in sage).

`lib.math.calculate_inverse_laplace()` now logs errors to `cwd()` as `log_file.txt` as well as `log_equations.txt` in case ilt is undefined or error occurs while `maxima` is running.
Same holds for `sum ETPs !=1`

**Comment by DRL** [comment]
Do we want to close this?

**Comment by DRL** [comment]
yes

**Comment by GertjanBisschop** [comment]
fixed in commit 98ea072c83385c65e2978233b0130d52be7a65cc
